vm/option.go

```diff
@@ -13,20 +16,38 @@ import (
 
 type RegisterFunc func(*VM)
 
-type OptionFunc func(*VM, []byte) error
+type OptionFunc func(vm *VM, configBytes []byte) error
```

Comment:
Is there ever a reason we want to expose this to the VM? Should we just hide this type within the package and force everyone to define a config per-option? Looks like we're using this for `WithManual` but I think we can just use a an empty struct for that config since it doesn't have one.

Reply:
We can un-export this, but the VM does need an option type that is ignorant of the generic type parameter. Then each option can call `NewOption[T]` with its own typed config and it receives a value of that type when it's executed.

Reply:
We could switch from `[]byte` to `interface{}`, so that it can unmarshal into that.

I'll just un-export this for now, but we do need a layer of indirection to hide the generic type from the VM.